plugin/src/main/kotlin/org/gradleweaver/plugins/jlink/JLinkLauncherOptions.kt

```diff
@@ -0,0 +1,48 @@
+package org.gradleweaver.plugins.jlink
+
+import java.io.Serializable
+
+class JLinkLauncherOptions : Serializable {
+
+    /**
+     * Runtime options for the JVM.
+     */
+    var vmOptions = mutableListOf<String>()
+
+    /**
+     * The name of the generated launcher script. If not set, the launcher name will be the name of the
+     * corresponding project.
+     */
+    var launcherName: String? = null
+
+    /**
+     * The name of the application module. Only needs to be set if the application is modular _and_ all dependencies
+     * are modular.
+     */
+    var applicationModuleName: String? = null
+
+    /**
+     * The fully qualified name of the main application class. Only needs to be set if the application is modular
+     * _and_ all dependencies are modular.
+     */
+    var mainClassName: String? = null
+
+    fun setVmOptions(options: Iterable<String>) {
+        vmOptions = options.toMutableList()
+    }
```

Comment:
Why have this extra method?

Reply:
So VM options can be more easily copied from the `application` plugin configuration:

```kotlin
application {
  applicationDefaultJvmArgs = listOf(...)
}

jlink {
  "configuration" {
    vmOptions(application.applicationDefaultJvmArgs)
  }
}
```

Reply:
Isn't that exactly the same as `vmOptions = application.applicationDefaultJvmArgs`?
Also, if you want that, this should really be:
```kotlin
vmOptions { application.applicationDefaultJvmArgs }
```
It would be better to make the `vmOptions` a `ListProperty`. That way the user can do:
```kotlin
vmOptions.set(provider { application.applicationDefaultJvmArgs} )
```

Reply:
That precludes users from adding more than just the JVM args from the application plugin, eg

```kotlin
jlink {
  "configuration" {
    vmOptions(application.applicationDefaultJvmArgs)
    vmOptions("-Xms16m", "-Xmx2G")
  }
}
```

Reply:
Not if using a `ListProperty`. `ListProperty` has methods like `addAll`.